source/initial_temperature/interface.cc

```diff
@@ -124,18 +134,39 @@ namespace aspect
 
       // go through the list, create objects and let them parse
       // their own parameters
-      for (unsigned int name=0; name<model_names.size(); ++name)
+      for (unsigned int i=0; i<model_names.size(); ++i)
         {
+          // create initial temperature objects
           initial_temperature_objects.push_back (std_cxx11::shared_ptr<Interface<dim> >
                                                  (std_cxx11::get<dim>(registered_plugins)
-                                                  .create_plugin (model_names[name],
+                                                  .create_plugin (model_names[i],
                                                                   "Initial conditions::Model names")));
 
           if (SimulatorAccess<dim> *sim = dynamic_cast<SimulatorAccess<dim>*>(&*initial_temperature_objects.back()))
             sim->initialize_simulator (this->get_simulator());
 
           initial_temperature_objects.back()->parse_parameters (prm);
           initial_temperature_objects.back()->initialize ();
+
+          // create operator list
+          {
+            unsigned int j = 0;
+            if (model_operator_names.size() != 1)
+              j = i;
```

Comment:
could this be done by using the `Utilities::possibly_extend_from_1_to_N` in the call to `prm.get...` like it is done in `material_model/multicomponent.cc`